### PR TITLE
team crud is completed

### LINKED ISSUE #22: Team
- Routes with CRUD operations and json response. 
- Models 
- Migrations. 

using Expressjs and Sequelize.

### PR DESCRIPTION
**Pull request by ismail-raza**

### CODE REVIEW

File: .idea/mean-stack.iml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (isheraz) [inline .idea/mean-stack.iml]: Add this file to gitignore

File: .idea/workspace.xml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (isheraz) [inline .idea/workspace.xml]: All IDE related files should always be ignored.

File: config/config.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require('dotenv').config();
```
Comment (isheraz) [inline config/config.js]: Please Rebase with Staging and use the configuration from there.

**Review by isheraz — CHANGES_REQUESTED** [review]

File: .idea/workspace.xml (hunk 2 above)
Comment (isheraz) [reply .idea/workspace.xml]: Remove this as well, please.

**Review by isheraz — CHANGES_REQUESTED** [review]

**Comment by isheraz** [comment]
Please see the conflicts and resolve those.

**Comment by ismail-raza** [comment]
> Please see the conflicts and resolve those.

OK

File: app.js [hunk 4]
```diff
@@ -12,6 +13,7 @@ require('dotenv').config()
  
 app.use('/role', checkRole('Super-Admin'), role);
 app.use('/permission',  permission);
+app.use('/team', team);
 //  checkPermission('Users'),
```
Comment (isheraz) [inline app.js]: No commented code should be part of the commit without a // TODO

File: constants.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 
 const PG_HOST = process.env.PG_HOST || '127.0.0.1';
-const PG_PORT = process.env.PG_PORT || 45775;
+const PG_PORT = process.env.PG_PORT || 5432;
```
Comment (isheraz) [inline constants.js]: I believe the default port is 45775, please revert and do not change the default port

File: controllers/team.controller.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+exports.getTeam = async (req, res) => {
+    try {
+        const team = await Team.findAll();
+        if (team) {
+            defaultResponse().success(constants.DATA_RETRIEVED, team, res, responseStatus.SUCCESS);
+        } else
+           defaultResponse().success(constants.DATA_NOT_FOUND, null, res, responseStatus.SUCCESS);
+    } catch (exception) {
+        defaultResponse().error({message:exception.message}, res, responseStatus.ERROR)
```
Comment (isheraz) [inline controllers/team.controller.js]: Please ensure the message and error sent here as a response is known and has an error code.

File: controllers/team.controller.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
+const Team = require('../models').Team;
+
+const defaultResponse= require('../utils/defaultResponse');
+const constants= require('../utils/constants');
+const responseStatus= require('../utils/responseStatus');
+
+exports.getTeam = async (req, res) => {
+    try {
+        const team = await Team.findAll();
+        if (team) {
```
Comment (isheraz) [inline controllers/team.controller.js]: checking for object validation should always be 
`if (!team) return res.status(404).message('team with id not found')` or similar

File: controllers/team.controller.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        }
+    }catch(exception){
+        defaultResponse().error({message:exception.message}, res, responseStatus.ERROR)
+    }
+}
+
+exports.delete= async(req,res)=>{
+    try{
+        const team= await Team.destroy({where:{id:req.params.id}})
+        if (team) {
```
Comment (isheraz) [inline controllers/team.controller.js]: similar for all requests. 
Also please add validation for each param.

File: models/Team.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        updatedAt:{type: DataTypes.DATE, defaultValue: Date.now}
+    }, {
+      sequelize,
+      modelName: 'Team',
+      paranoid: true,
+    });
+    return Team;
+  };
+  
+// const Sequelize = require('sequelize');
```
Comment (isheraz) [inline models/Team.js]: Remove Comments

File: package.json [hunk 10]
```diff
@@ -40,11 +40,7 @@
     "lint-staged": "^10.5.3",
     "prettier": "2.2.1"
   },
-  "husky": {
```
Comment (isheraz) [inline package.json:53]: Please do not remove `husky` or pre-commit actions, add these back.

**Review by isheraz — CHANGES_REQUESTED** [review]

**Comment by isheraz** [comment]
Also, delete and create a new branch from origin since I removed the .idea directory for you. additionally, rebase all your commits into one.

**Review by isheraz — CHANGES_REQUESTED** [review]
Please Make Sure you run 
`git branch -D staging && git fetch --all && git checkout -b staging origin/staging && git checkout team && git rebase staging` and resolve all conflicts with latest changes.

File: _helpers/basicAuth.js [hunk 11]
```diff
@@ -30,20 +31,25 @@ const checkPermission = (permission) => async (_req, res, next) => {
   });
   const permissions = role ? role.permissions : null;
 
-  const isPermission = permissions.some(
-    (permittedUser) => permittedUser.name === permission
-  );
+  // if (typeof userPermission === 'string') {
+  //   userPermission[] = [userPermission];
```
Comment (isheraz) [inline _helpers/basicAuth.js]: Remove Comments

File: migrations/20201223081551-team.js [hunk 12]
```diff
@@ -1,3 +1,26 @@
+<<<<<<< HEAD
```
Comment (isheraz) [inline migrations/20201223081551-team.js]: Fix this merge issue

File: migrations/20201223081551-team.js [hunk 13]
```diff
@@ -1,3 +1,26 @@
...
+    up: async (queryInterface, Sequelize) => {
+
+        return queryInterface.createTable('Teams', {
+            id: {
+                type: Sequelize.INTEGER,
+                autoIncrement: true,
+                primaryKey: true,
+            },
+            name: Sequelize.STRING,
+            members: Sequelize.STRING,
```
Comment (isheraz) [inline migrations/20201223081551-team.js]: Adjust as needed for grouping team

**Review by isheraz — CHANGES_REQUESTED** [review]

**Pull request merged**